modules/scanning-delegate-role/main.tf

```
@@ -539,7 +542,146 @@ resource "aws_iam_role_policy_attachment" "worker_attachment" {
 }
 
 resource "aws_iam_role_policy_attachment" "workers_dspm_attachment" {
-  count      = var.sensitive_data_scanning_enabled ? 1 : 0
+  count      = length(aws_iam_policy.scanning_worker_dspm_policy)
   policy_arn = aws_iam_policy.scanning_worker_dspm_policy[0].arn
   role       = aws_iam_role.role.name
 }
+
+// RDS Specific resources
+
+// RDS Service Role for S3 Exports
+data "aws_iam_policy_document" "rds_service_role_assume_policy" {
+  statement {
+    actions = ["sts:AssumeRole"]
+    effect  = "Allow"
+    principals {
+      type        = "Service"
+      identifiers = ["export.rds.amazonaws.com"]
+    }
+  }
+}
+
+data "aws_iam_policy_document" "rds_service_role_policy_document" {
+  statement {
+    sid = "DatadogAgentlessScannerExportPolicy"
```

```suggestion
    sid = "DatadogAgentlessScannerRdsExportPolicy"
```

(if this is specific to RDS)